cmd/certificate_maker/certificate_maker.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package main implements a certificate creation utility for Fulcio.
+// It supports creating root and leaf certificates using (AWS, GCP, Azure).
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/sigstore/fulcio/pkg/certmaker"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+// CLI flags and env vars for config.
+// Supports AWS KMS, Google Cloud KMS, and Azure Key Vault configurations.
+var (
+	logger  *zap.Logger
+	version string
+
+	rootCmd = &cobra.Command{
+		Use:     "fulcio-certificate-maker",
+		Short:   "Create certificate chains for Fulcio",
+		Long:    `A tool for creating root, intermediate, and leaf certificates for Fulcio with code signing capabilities`,
+		Version: version,
+	}
+
+	createCmd = &cobra.Command{
+		Use:   "create",
+		Short: "Create certificate chain",
+		RunE:  runCreate,
+	}
+
+	kmsType              string
+	kmsRegion            string
+	kmsKeyID             string
+	kmsTenantID          string
+	kmsCredsFile         string
+	rootTemplatePath     string
+	leafTemplatePath     string
+	rootKeyID            string
+	leafKeyID            string
+	rootCertPath         string
+	leafCertPath         string
+	intermediateKeyID    string
+	intermediateTemplate string
+	intermediateCert     string
+	kmsVaultToken        string
+	kmsVaultAddr         string
+
+	rawJSON = []byte(`{
```

Comment:
I'd define this locally within the method it is used as a native struct; I don't see the benefit of unmarshalling it from JSON

Reply:
ah, thanks for catching that.